seeker/host/disk.go

```diff
@@ -0,0 +1,29 @@
+package host
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcdiag/seeker"
+	"github.com/shirou/gopsutil/v3/disk"
+)
+
+var _ seeker.Runner = Disk{}
+
+type Disk struct{}
+
+func NewDisk() *seeker.Seeker {
+	return &seeker.Seeker{
+		Identifier: "disks",
+		Runner:     Disk{},
+	}
+}
+
+func (dp Disk) Run() (interface{}, seeker.Status, error) {
+	// third party
+	diskInfo, err := disk.Partitions(true)
+	if err != nil {
+		return diskInfo, seeker.Unknown, fmt.Errorf("error getting disk information err=%w", err)
```

Comment:
It looks like the other seekers in this PR will write a log message and then return `<info>, seeker.Fail, err`. Should we do that here too?

Reply:
Ah yeah... that's a good catch. This is a good nudge for me to go to go back and change all of the host seekers to match this form, rather than following the old form we've been copy-pasting around. Users will get a warn-level log if a seeker returns an error, so we don't need the Error logs in the `seeker/host` Run()s. If anything it's misleading because it shouldn't require action on the users' part.

Reply:
Changed these to traces that name the method directly 🚢